src/utils/data_transform.py

```diff
@@ -105,7 +112,23 @@ def transform(self, image_stack, mask=None, return_unnormalized_numpy=False):
         if return_unnormalized_numpy:
             return image_stack, mask
 
-        return self.normalize_and_torchify(image_stack, mask)
+        image_stack, mask = self.normalize_and_torchify(image_stack, mask)
+
+        if self.is_train and self.temporal_augmentation and np.random.rand() < 0.8:
+            assert (
+                len(image_stack.shape) == 3
+            ), f"Expecting an image stack with shape (Temporal, Bands, Pixels) but got shape {image_stack.shape}"
+
+            synthetic_image_stack = np.zeros(image_stack.shape)
+            # Go through pixel
+            for i in image_stack.shape[2]:
```

Comment:
Do you mean:

for i in range(image_stack.shape[2]): 

?

Reply:
yes, thank you, fixed!